src/components/NoteForm.js

```diff
@@ -10,14 +10,18 @@ import ReactMarkdown from "react-markdown";
 import api from "../lib/apiGraphQL";
 
 function NoteForm({goalId, repoName, note}) {
+  const [previouslySavedValue, setPreviouslySavedValue] = useState(note);
```

Comment:
Needed to save this off since otherwise you'd end up reverting back to the version from page load (`note`) instead of the last saved version